src/routing/Router.svelte

```diff
@@ -49,28 +49,29 @@
     return function handle({ params: { product, section, probeName } }) {
       const storeValue = get(store);
       component = componentToUse;
-      // if the product has changed,
-      // set it in the store and use store.resetProductDimensions()
-      // to initialize.
-      if (product && storeValue.product !== product) {
-        store.setField('product', product);
-        store.resetProductDimensions();
-      }
+
       // Issue #355: Update the probe here, whenever the path changes, to ensure
       // that clicks to the back/forward buttons work as expected.
-
       if (probeName) {
+        store.setField('probe', { loaded: false });
         store.setField('probeName', probeName);
 
         // The canonical probe info fetch. (PSS)
-        getSearchResults(probeName, true).then((r) => {
+        getSearchResults(probeName, true, $store.searchProduct).then((r) => {
           let newProbe = { ...r[0], loaded: true };
+
+          // if the product has changed,
+          // set it in the store and use store.resetProductDimensions()
+          // to initialize.
+          if (product && storeValue.product !== product) {
+            store.setProduct(product);
+          }
```

Comment:
In general, I'm a bit concerned about *Router.svelte* becoming too complex. Maybe some of this could be moved elsewhere in the future?